nb2/bot/socket_mode.py

```diff
@@ -0,0 +1,29 @@
+from flask import Blueprint
+from slack_sdk.socket_mode import SocketModeClient
+from slack_sdk.socket_mode.request import SocketModeRequest
+from slack_sdk.socket_mode.response import SocketModeResponse
+
+from nb2 import bot
+
+bp = Blueprint("slack_socket_mode", __name__)
+
+
+class FlaskSocketModeClient(SocketModeClient):
+    def __init__(self, app, app_token, web_client):
+        super().__init__(app_token=app_token, web_client=web_client)
+        self.app = app
+        self.socket_mode_request_listeners.append(self.process)
+        self.connect()
+
+    def process(self, client: SocketModeClient, req: SocketModeRequest):
+        if req.type == "events_api":
```

Comment:
To get rid of the indenting, I normally write stuff like this as:

```
if req.type != "events_api":
    return

# Acknowledge the request anyway
....
```

Reply:
updated